bundles/org.openhab.binding.buienradar/src/main/java/org/openhab/binding/buienradar/internal/BuienradarHandler.java

```diff
@@ -133,9 +171,14 @@ private void refresh() {
     @SuppressWarnings("null")
     @Override
     public void dispose() {
-        if (listenableFuture != null && !listenableFuture.isCancelled()) {
-            listenableFuture.cancel(true);
-            listenableFuture = null;
+        try {
+            listenableFutureLock.lock();
```

Comment:
I don‘t think this hurts, but: why did you add the lock here?

Reply:
Because I cannot sync on a nullable value (listenableFuture) and this makes the class thread-safe (initialize / dispose both read/write listenableFuture).